app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
+        self.wife = None
+        self.husband = None
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_people = []
+    for data in people:
+        name = data["name"]
+        age = data["age"]
+        person = Person(name, age)
+        list_of_people.append(person)
+
+    for person in list_of_people:
+        if person.name in Person.people:
+            data = next((d for d in people if d["name"] == person.name), None)
+            if "wife" in data and data["wife"] is not None:
```

Comment:
You can simplify your code and use method `get` of `dict`
```suggestion
            if data.get("wife"):
```